simplify-basedomain-usage/README.md

```diff
@@ -0,0 +1,175 @@
+
+###### dns-operator-*
+
+It needs it to create the hosted zone and DNS records for the cluster that's reconciling.
+
+###### irsa-operator
+
+For CAPI clusters it loads the `cluster-values` configmap from the management cluster. It should also take the value from the annotation instead.
+
+## Proposal #2: Stop using the `baseDomain` from the `cluster-values` configmap in CAPI installations
+
+The `cluster-apps-operator` in CAPI installations, (and the `cluster-operator` in `vintage` installations) is creating a configmap called `$clustername-cluster-values` that contains values related to the workload cluster called `$clustername`.
+One of these values is the `baseDomain` for that cluster.
+This configmap is then added as [a config source in the `App` CRs created for the workload clusters default apps](https://github.com/giantswarm/default-apps-aws/blob/master/helm/default-apps-aws/templates/apps.yaml#L38-L40).
+
+In CAPI installations, when installing the default apps in the `default-apps-$provider` app bundle, we can pass configuration values, [like we currently do for `cilium`](https://github.com/giantswarm/default-apps-aws/blob/master/helm/default-apps-aws/values.yaml#L10-L26).
+This means that we can pass the `baseDomain` to the default apps that need it. There is no need to add a reference to the `cluster-values` configmap to the `App` CR of the default apps.
+
+### Advantages
+
+#### More explicit, less surprises
+
+This way we know exactly which default apps depend on the `baseDomain` value, and we explicitly pass it, instead of relying on some mechanism that automagically passes the value.
+
+#### Reduce coupling
+
+Applications fetching this value from the `cluster-values` configmap depend on the `cluster-apps-operator` or `cluster-operator` being deployed and creating the configmap.
+Using the value directly from the `default-apps-$provider` bundle instead of depending on the configmap reduces coupling between our apps.
+
+#### Reduce complexity
+
+We would completely eliminate one of the ways of configuring the `baseDomain`, by removing it from the `cluster-values` configmap.
+
+#### Aligned with our GitOps strategy
+
+All the configuration would be in the repository containing the default apps manifests, instead of this configmap being dynamically created by an operator.
+
+#### No need to migrate cluster-apps-operator
+
+If `cluster-aps-operator` wouldn't need the `baseDomain` anymore, we wouldn't need to migrate the operator to take the value from an annotation.
+
+### Downsides
+
+#### Verbosity in default apps bundle
+
+We would need to explicitly pass the `baseConfig` to all the different apps that need it in the default apps bundle.
+
+##### List of default apps using the `baseDomain` from the `cluster-values` configmap
+
+###### cilium
+
+It uses the `baseDomain` to talk to the k8s api. It already receives the `baseDomain` from the `default-apps-$provider` bundle, so there is no need to do anything here.
```

Comment:
Overall, can we even bring down the usage of `baseDomain` to **either** proposal 1 or 2, removing all other usages? I feel we must keep living with a mixture, or proposal 2 alone.

I don't see how proposal 1 (add a `Cluster` annotation) would allow our app Helm templates to read that value – unless we use Helm's [`lookup` function](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-lookup-function) which IMO raises the level of dependencies and mental complexity.